var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -51,6 +51,7 @@ class Paraview(Package):
     depends_on('qt@:4', when='+qt')
 
     depends_on('cmake', type='build')
+    depends_on('mesa')
```

Comment:
I do not think we should not have it unconditional: 
As far as I know, If your system has GPU, you should use "native" Opengl and not Mesa
Is not there  already an osmesa variant in ParaView?
Also did you compile Qt in spack or use the system provided?
I have the need to install Paraview + Qt on two cluster, one with GPU and another without.
On the one with I think that OpenGL is ( and has to be ) system installed, while on the one without, I nedd to install Mesa, and installed one with Gallium, as per 
https://github.com/RemoteConnectionManager/spack/tree/wip/mesa_gallium
It is not ready but maybe worthwhile a WIP PR